### PR TITLE
 add r2 (partial) support

### PR DESCRIPTION
**Pull request by ryoppippi**
Hi!
Added support for some r2 instructions.
I confirmed it is working at hand.
Please take a look if you like.

### CODE REVIEW

**Comment by ryoppippi** [comment]
I tested in my environment and I think it is ok to work

**Comment by leaysgur** [comment]
Thanks again.

I haven't been able to test it yet, and there are a few things I would like to look into..., but I think I will merge for the time being.

File: .DS_Store
Comment (leaysgur) [inline .DS_Store:1]: Would you please delete this and update `.gitignore`?

File: bridge/r2.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+  /** @param {import("@cloudflare/workers-types").R2ListOptions} [options] */
+  async list(options) {
+    const url = new URL(this.#wranglerOrigin);
+    url.pathname = `/r2_list/${this.#bindingName}`;
+
+    const req = new Request(url, { method: "GET" });
+    if (options) req.headers.set("CF-R2-OPTIONS", JSON.stringify(options));
+
+    return await fetch(req);
```
Comment (leaysgur) [inline bridge/r2.js]:
```suggestion
    const res = await fetch(req);
    return res;
```

Align code styles. 💅🏻

**Comment by ryoppippi** [comment]
Thanks...
I think we need to write some test maybe.

File: worker/r2.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const options = JSON.parse(req.headers.get("CF-R2-OPTIONS") ?? "{}");
+
+  if (OPERATION === "r2_head") {
+    const result = await R2.head(key);
+    return Response.json(result);
+  }
+
+  if (OPERATION === "r2_get") {
+    const result = await R2.get(key, options);
+    console.log({ result });
```
Comment (leaysgur) [inline worker/r2.js]:
```suggestion
```

**Review by leaysgur — CHANGES_REQUESTED** [review]

File: .DS_Store
Comment (ryoppippi) [reply .DS_Store:1]: OH sorry Thanks

**Comment by leaysgur** [comment]
> I think we need to write some test maybe.

Yep, at least something that can run locally. 😅

**Comment by ryoppippi** [comment]
I fixed!

> Yep, at least something that can run locally.

Yeah, as far as I tested on my real SveteKit Application, it looks good.
Maybe We'll write a simple test.


Again, thank you for creating this. 
I do not want to build the entire application in every 10 sec anymore....

**Comment by leaysgur** [comment]
> I do not want to build the entire application in every 10 sec anymore....

😭 

(Ultimately, I am hoping that people at @cloudflare will find a nice solution and this module will be a temporary thing.)

**Comment by leaysgur** [comment]
`0.2.0` published 🎁,  Thanks again!

**Comment by ryoppippi** [comment]
> (Ultimately, I am hoping that people at https://github.com/cloudflare will find a nice solution and this module will be a temporary thing.)

absolutely yes lol

**Comment by ryoppippi** [comment]
Thanks!!

**Pull request merged**